### PR TITLE
netstat: Simplify pod interface volatile cache

### PR DESCRIPTION
**Pull request by EdDev**
**What this PR does / why we need it**:
Simplify the existing code that implements the cache behind the pod interface, by removing several wrappers and optimizing some parts of the logic.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
~~Depends on #6781~~
~~Only the last 3 commits are relevant to review in this context.~~

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/network/setup/netstat.go [hunk 1]
```diff
@@ -38,27 +36,32 @@ type NetStat struct {
...
+		if strings.Contains(key.(string), string(vmi.UID)) {
+			c.podInterfaceVolatileCache.Delete(key)
+		}
+		return true
+	})
 }
 
 func (c *NetStat) PodInterfaceVolatileDataIsCached(vmi *v1.VirtualMachineInstance, ifaceName string) bool {
-	_, exists := c.podInterfaceVolatileCache.Load(vmi.UID, ifaceName)
+	_, exists := c.podInterfaceVolatileCache.Load(fmt.Sprintf("%s/%s", vmi.UID, ifaceName))
```
Comment (orelmisan) [inline pkg/network/setup/netstat.go:59]: This function call is repeated several times in the file, can you please consider adding a function that generates the key?

File: pkg/network/setup/netstat.go [hunk 2]
```diff
@@ -201,16 +204,15 @@ func (c *NetStat) UpdateStatus(vmi *v1.VirtualMachineInstance, domain *api.Domai
...
 	// If already present in the map, don't read again
-	podInterface, exists := c.podInterfaceVolatileCache.Load(vmi.UID, ifaceName)
-
+	cacheData, exists := c.podInterfaceVolatileCache.Load(fmt.Sprintf("%s/%s", vmi.UID, ifaceName))
 	if exists {
-		return podInterface, nil
+		return cacheData.(*cache.PodCacheInterface), nil
 	}
 
 	//FIXME error handling?
```
Comment (orelmisan) [inline pkg/network/setup/netstat.go:212]: Is this comment still relevant?

**Review by orelmisan — CHANGES_REQUESTED** [review]
Hi @EdDev, looks good, few minor issues to address.

File: pkg/network/setup/netstat.go (hunk 1 above)
Comment (EdDev) [reply pkg/network/setup/netstat.go:59]: Why do we need to wrap ‘fmt.Sprintf’ with another function?

The only advantage will be to skip the format string, and it is short and to the point that does not require explanation.

File: pkg/network/setup/netstat.go (hunk 2 above)
Comment (EdDev) [reply pkg/network/setup/netstat.go:212]: Yes, what changed in this regard?

File: pkg/network/setup/netstat.go (hunk 1 above)
Comment (orelmisan) [reply pkg/network/setup/netstat.go:59]: The advantage as you've said is to skip the format string which could be error prone if a space character slips.
Furthermore, a function call could also improve the code's readability IMO.

Comment (orelmisan) [reply pkg/network/setup/netstat.go]: Thank you

**Review by orelmisan — APPROVED** [review]

**Comment by EdDev** [comment]
change: Rebase

**Comment by EdDev** [comment]
@orelmisan , can you re-lgtm? I had to rebase.

**Review by orelmisan — APPROVED** [review]

**Pull request merged**